Algorithm.Framework/Execution/StandardDeviationExecutionModel.cs

```diff
@@ -0,0 +1,217 @@
+                }
+            }
+
+            foreach (var removed in changes.RemovedSecurities)
+            {
+                // clean up data from removed securities
+                SymbolData data;
+                if (_symbolData.TryGetValue(removed.Symbol, out data))
+                {
+                    if (IsSafeToRemove(algorithm, removed.Symbol))
+                    {
+                        _symbolData.Remove(removed.Symbol);
+                        algorithm.SubscriptionManager.RemoveConsolidator(removed.Symbol, data.Consolidator);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Determines if the current price is more than the configured number of standard deviations
+        /// away from the mean in the favorable direction.
+        /// </summary>
+        private bool PriceIsFavorable(SymbolData data, decimal unorderedQuantity)
+        {
+            var deviations = _deviations * data.STD;
+            if (unorderedQuantity > 0)
+            {
+                var price = data.Security.BidPrice == 0
+                    ? data.Security.Price
+                    : data.Security.BidPrice;
+
+                if (price < data.SMA - deviations)
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                var price = data.Security.AskPrice == 0
+                    ? data.Security.AskPrice
+                    : data.Security.Price;
+
+                if (price > data.SMA + deviations)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Determines if it's safe to remove the associated symbol data
+        /// </summary>
+        private bool IsSafeToRemove(QCAlgorithmFramework algorithm, Symbol symbol)
+        {
+            // confirm the security isn't currently a member of any universe
+            return !algorithm.UniverseManager.Any(kvp => kvp.Value.ContainsMember(symbol));
+        }
+
+        class SymbolData
+        {
+            public Security Security { get; }
+            public StandardDeviation STD { get; }
+            public SimpleMovingAverage SMA { get; }
+            public IDataConsolidator Consolidator { get; }
+
+            public SymbolData(QCAlgorithmFramework algorithm, Security security, int period, Resolution resolution)
+            {
+                Security = security;
+                Consolidator = algorithm.ResolveConsolidator(security.Symbol, resolution);
+                var smaName = algorithm.CreateIndicatorName(security.Symbol, "SMA" + period, resolution);
+                SMA = new SimpleMovingAverage(smaName, period);
+                var stdName = algorithm.CreateIndicatorName(security.Symbol, "STD" + period, resolution);
+                STD = new StandardDeviation(stdName, period);
```

Comment:
Later we can add a history call here to warm up the indicators

Reply:
added! along w/ helper method